src/controllers/game_controller.cpp

```diff
@@ -42,19 +42,25 @@ void GameController::createSystems() {
 }
 
 void GameController::createTestEntities() {
-    entityFactory->createPanda(400, 200);
-    entityFactory->createGorilla(1000, 200);
+    entityFactory->createPanda(400, 200, 1);
+    entityFactory->createGorilla(1000, 200, 2);
     entityFactory->createImage("backgrounds/forest_watermarked.jpg", SCREEN_WIDTH / 2, SCREEN_HEIGHT / 2, SCREEN_WIDTH, SCREEN_HEIGHT, Layers::Background);
     entityFactory->createPlatform(640, 500, 720, 20);
 }
 
 void GameController::setupInput() {
     BrickInput<PlayerInput>& input = BrickInput<PlayerInput>::getInstance();
-    std::unordered_map<SDL_Keycode, PlayerInput> inputMapping;
-    inputMapping[SDLK_w] = PlayerInput::PLAYER1_UP;
-    inputMapping[SDLK_a] = PlayerInput::PLAYER1_LEFT;
-    inputMapping[SDLK_s] = PlayerInput::PLAYER1_DOWN;
-    inputMapping[SDLK_d] = PlayerInput::PLAYER1_RIGHT;
+    std::unordered_map<int, std::unordered_map<SDL_Keycode, PlayerInput>> inputMapping;
+    // Player 1
+    inputMapping[1][SDLK_w] = PlayerInput::UP;
```

Comment:
The Game now still "knows" about the SDL inputs. I think you should create a second enum/inputmapping that maps keyboard inputs from the engine to SDL

Reply:
Shouldnt be too hard, will look at it tomorrow